terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -210,29 +210,85 @@ def source_type(source_string)
       end
       # rubocop:enable Metrics/PerceivedComplexity
 
+      def parsed_file_hcl2(file)
+        SharedHelpers.in_a_temporary_directory do
+          File.write("tmp.tf", file.content)
+
+          command = "#{terraform_hcl2_parser_path} < tmp.tf"
+          start = Time.now
+          stdout, stderr, process = Open3.capture3(command)
+          time_taken = Time.now - start
+
+          unless process.success?
+            raise SharedHelpers::HelperSubprocessFailed.new(
+              message: stderr,
+              error_context: {
+                command: command,
+                time_taken: time_taken,
+                process_exit_value: process.to_s
+              }
+            )
+          end
+
+          JSON.parse(stdout)
+        end
+      end
+
+      def parsed_file_hcl1(file)
+        SharedHelpers.in_a_temporary_directory do
+          File.write("tmp.tf", file.content)
+
+          command = "#{terraform_parser_path} -reverse < tmp.tf"
+          start = Time.now
+          stdout, stderr, process = Open3.capture3(command)
+          time_taken = Time.now - start
+
+          unless process.success?
+            raise SharedHelpers::HelperSubprocessFailed.new(
+              message: stderr,
+              error_context: {
+                command: command,
+                time_taken: time_taken,
+                process_exit_value: process.to_s
+              }
+            )
+          end
+
+          json = JSON.parse(stdout)
+          json["module"] = json.fetch("module", []).inject({}) { |memo, item| memo.merge(item) }
+          json
+        end
+      end
+
+      # == Returns:
+      # A Hash representing each module found in the specified file
+      #
+      # E.g.
+      # {
+      #   "module" => {
+      #     {
+      #       "consul" => [
+      #         {
+      #           "source"=>"consul/aws",
+      #           "version"=>"0.1.0"
+      #         }
+      #       ]
+      #     }
+      #   },
+      #   "terragrunt"=>[
+      #     {
+      #       "include"=>[{ "path"=>"${find_in_parent_folders()}" }],
+      #       "terraform"=>[{ "source" => "git::git@github.com:gruntwork-io/modules-example.git//consul?ref=v0.0.2" }]
+      #     }
+      #   ],
+      # }
       def parsed_file(file)
         @parsed_buildfile ||= {}
         @parsed_buildfile[file.name] ||=
-          SharedHelpers.in_a_temporary_directory do
```

Comment:
Nit - should the `SharedHelpers.in_a_temporary_directory` block stay in this method?

It's not a big deal as there's a lot of repetition in the two `parse_file_hcl*` methods and trying to DRY that out would reduce clarity IMO. I think I've talked myself out of it?